### PR TITLE
MSAL support

### LINKED ISSUE #10259: Move to MSAL
## Description

We are moving from ADAL to MSAL

- [x] See if there is anywhere else needs tracing message
- [ ] Cancellation token

## How to verify
- [x] Verify all auth scenarios
- [x] Verify token cache migration
- [x] Verify tracing message
    - [x] No sensitive info is in the messages
- [ ] Verify SSO with lastest VS build
- [ ] Verify cancellation token (by unit tests)

## Release

- [ ] Release as 1.7.0
- [ ] After release, hide 2.0.0-preview

## Cost

10

A relevant issue on ADFS #10260

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

Fixes #10259 #10260

What's *not* in:
- Version update of Az.Accounts (I plan to do it in aother PR)
- Token cache migration
- Utilize official build of common code instead of local packages

## Reference

Original PR https://github.com/Azure/azure-powershell/pull/9982

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by isra-fel** [comment]
I'll try fixing the test cases tomorrow

File: src/Accounts/Authenticators/AccessTokenAuthenticator.cs [hunk 1]
```diff
@@ -41,21 +41,27 @@ public override Task<IAccessToken> Authenticate(AuthenticationParameters paramet
             };
 
             if ((resourceId.EqualsInsensitively(environment.AzureKeyVaultServiceEndpointResourceId) ||
-                 resourceId.EqualsInsensitively(AzureEnvironment.Endpoint.AzureKeyVaultServiceEndpointResourceId))
+                 resourceId.EqualsInsensitively(AzureEnvironment.Endpoint.AzureKeyVaultServiceEndpointResourceId) ||
```
Comment (markcowl) [inline src/Accounts/Authenticators/AccessTokenAuthenticator.cs]: So, it looks like the common change in behavior is to allow the resourceID parameter to either match the actual resourceId, or to match the endpoint name that contains the resource ID, which is the correct behavior.

Alternately, we could normalize the AuthenticationParameters to always use one form or the other.

Either way, we need a regression test in place to ensure we always do this.  How difficult would it be to write a unit test for this, or add this to an existing unit test?

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
Closing this one as MSAL code has been checked in to `Azs-tzl`

**Pull request closed without merging**